runtime/src/accounts_index.rs

```diff
@@ -685,9 +685,11 @@ impl<T: 'static + Clone + IsCached + ZeroLamport> AccountsIndex<T> {
     }
 
     pub fn get_rooted_entries(&self, slice: SlotSlice<T>, max: Option<Slot>) -> SlotList<T> {
+        let max = max.unwrap_or(Slot::MAX);
+        let lock = &self.roots_tracker.read().unwrap().roots;
         slice
             .iter()
-            .filter(|(slot, _)| self.is_root(*slot) && max.map_or(true, |max| *slot <= max))
+            .filter(|(slot, _)| *slot <= max && lock.contains(slot))
```

Comment:
could also check `slot <= roots_tracker.max_root` before checking the `.contains()`. Not sure if that would be faster or not.

Reply:
contains will shortly be low constant time, so I think the change wouldn't help and would (likely very slightly) slow down the case when contains returns true.